ui/src/shared/components/editors/graph-viewer.tsx

```diff
@@ -0,0 +1,356 @@
+import * as React from 'react';
+import {useEffect, useState} from 'react';
+
+import {genres} from '../../../workflows/components/workflow-dag/genres';
+import {WorkflowDagRenderOptions} from '../../../workflows/components/workflow-dag/workflow-dag';
+import {WorkflowDagRenderOptionsPanel} from '../../../workflows/components/workflow-dag/workflow-dag-render-options-panel';
+import {ClusterWorkflowTemplate, CronWorkflow, DAGTask, Template, Workflow, WorkflowStep, WorkflowTemplate} from '../../models';
+import {services} from '../../services';
+import {GraphPanel} from '../graph/graph-panel';
+import {Graph} from '../graph/types';
+import {Icon} from '../icon';
+
+export function GraphViewer({workflowDefinition}: {workflowDefinition: Workflow | WorkflowTemplate | ClusterWorkflowTemplate | CronWorkflow}) {
+    const [workflow, setWorkflow] = useState<Workflow | WorkflowTemplate | ClusterWorkflowTemplate>(workflowDefinition);
+    const [isLoading, setIsLoading] = useState(true);
+    const [state, saveOptions] = useState<WorkflowDagRenderOptions>({
+        expandNodes: new Set(),
+        showArtifacts: false,
+        showInvokingTemplateName: false,
+        showTemplateRefsGrouping: false
+    });
+    const defaultNodeParams: {
+        classNames: string;
+        progress: number;
+        icon: Icon;
+    } = {
+        classNames: 'Skipped',
+        progress: 0,
+        icon: 'clock'
+    };
+
+    useEffect(() => {
+        if ('workflowTemplateRef' in workflowDefinition.spec && isLoading) {
+            setWorkflowFromRefrence(workflowDefinition.spec.workflowTemplateRef.name).then(() => {
+                setIsLoading(false);
+            });
```

Comment:
cluster workflow template use case is missing
https://github.com/argoproj/argo-workflows/tree/main/examples/cluster-workflow-template